src/com/nkita/practice/collections/PersonStatisticHandler.java

```diff
@@ -176,27 +170,59 @@ public void checkFullStatisticWithoutDuplicates() {
         Set<Person> personSet = new LinkedHashSet<>(persons);
         System.out.println("Statistics with duplicate");
 
-        for (Person person : persons) {
-            personSet.add(person);
-            System.out.println(personSet);
-            return;
+        for (Person person : personSet) {
+            System.out.println(person);
         }
 
 
     }
 
 
-
     public void removeDuplicates() {
         Set<Person> personSet = new LinkedHashSet<>(persons);
         persons.clear();
         persons.addAll(personSet);
 
-        }
+    }
+
+    private List<Person> getPersonsListByAge(int age, Gender gender) {
+
+        List<Person> personsList = new ArrayList<>();
+
+        for (Person person : persons) {
+            if (person.getAge() == age && (gender == null || gender == person.getGender())) {
+                personsList.add(person);
+            }
 
+        }
+        return personsList;
 
     }
 
 
+    public void checkOldestPerson(Gender gender) {
 
+        int maxAge = 0;
+        for (Person person : persons) {
+            if (person.getAge() > maxAge) {
+
+                if (gender == null || person.getGender() == gender) {
+                    maxAge = person.getAge();
+                }
+            }
+        }
 
+        List<Person> genderMaleAgeAndGenderFemaleAge = getPersonsListByAge(maxAge, gender);
+
+        if (gender == null) {
```

Comment:
"gender == null"  is duplicate  from line 209
 Please solve the problem with duplicates.